File: 392-is-subsequence/is-subsequence.kt
class Solution {
    fun isSubsequence(s: String, t: String): Boolean {
        
        //if no string in t
        if(t.length == 0){
            if(s.length == 0) return true
            else return false
        }

        //if no string in s
        if(s.length == 0) return true


        var i = 0
        var j = 0

        while (i < t.length && j < s.length){
            if(t[i] == s[j]) j++
            i++
        }

        if(j == s.length) return true

        return false
    }
}

File: 392-is-subsequence/is-subsequence.py
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) == 0:
            return True
        i = 0
        for ch in t:
            if ch == s[i]:
                i += 1
            if len(s) == i:
                return True
        return False   
        